library/std/src/sys/itron/time.rs

```diff
@@ -0,0 +1,123 @@
+use super::{abi, error::ItronError};
+use crate::{convert::TryInto, mem::MaybeUninit, time::Duration};
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct Instant(abi::SYSTIM);
```

Comment:
Is it allowed for `Instant` to have a granularity bigger than nanoseconds? It is observable when substracting or adding nanosecond durations.

Reply:
The Darwin targets use a Mach timebase, which is not guaranteed to be constant. Also, [the docs](https://doc.rust-lang.org/std/time/struct.Instant.html#os-specific-behaviors) says:

> An `Instant` is a wrapper around system-specific types and it may behave differently depending on the underlying operating system.